### PR TITLE
Briefings: bottom share drawer backed by server-side PDF

### PR DESCRIPTION
**Pull request by oakinh**
## Summary
- Replaces the desktop "Download" header button and the mobile dock's "Download" FAB with a single "Share" entry point that opens a bottom-anchored share drawer.
- The drawer surfaces four circular actions — Copy link, Email (`mailto:`), Message (`sms:`), Download — plus an inline URL pill with a Copy button. Every action targets the same public PDF URL, served by gp-api via a new Vercel rewrite (`/api/v1/briefings/:uuid` → `${NEXT_PUBLIC_API_BASE}/v1/briefings/:uuid`).
- Drops the client-side `@react-pdf/renderer` code path (`BriefingPdfDocument.tsx`, `downloadBriefingPdf.tsx`) and uninstalls the package (-47 transitive deps). The server-side `pdfkit` renderer in gp-api is now the single source of truth.

## Architecture
- `ShareScope` provider at `[slug]/layout.tsx` owns the only `<ShareBriefingDrawer>` instance on the page; `DetailHeaderActions` and `MobileBottomBar` consume `useShareScope()` to dispatch the open call. Eliminates the duplicate Radix `<Sheet>` mount and the focus-trap race between them after the breakpoint flips.
- Copy-feedback `setTimeout`s tracked in refs and cleared on unmount; rapid double-clicks reset the timer rather than stacking handlers.
- New `formatBriefingMeetingDate` / `formatBriefingMeetingTime` helpers in `@shared/briefings/dateHelpers` so the drawer subtext and the gp-api header line agree on shape.

## Type-contract drift
- `Briefing` was extended with `briefing_id`, `meeting_time?`, and `meeting_timezone?` (the `MeetingBriefingFull` contract is stale; doc-commented inline). `isFullArtifact` is hardened with a runtime `briefing_id` check so a deploy where gp-api hasn't shipped the augmented payload yet won't generate `/api/v1/briefings/undefined` links.
- `next.config.ts` fails loudly and skips the rewrite when `NEXT_PUBLIC_API_BASE` is unset, preventing silent 404s on the share URL.

## Test plan
- [x] `npx vitest run app/dashboard/briefings/components` — 22 files / 179 tests passing.
- [x] `npx tsc --noEmit` clean.
- [x] eslint clean on all touched files.
- [ ] Manual: with the gp-api PR deployed, open a briefing, click Share, copy the link, paste in a private window — PDF should stream from `${APP_BASE}/api/v1/briefings/<uuid>`.
- [ ] Manual: confirm `mailto:` opens the user's mail client with subject + body pre-filled; `sms:?body=` opens the messaging app on iOS/Android.
- [ ] Manual: verify the Download action streams the PDF inline / saves with the slugified filename from the server.

## Test quality changes (per critic review)
- Split the previously multi-assertion drawer render test into focused cases.
- Subtext check uses the exact `"City Council · Mon May 11 · 6:00 PM · City Hall Council Chambers"` string instead of a fuzzy regex.
- New clipboard-rejection test verifies the UI does not falsely claim a successful copy when `navigator.clipboard.writeText` throws.
- Dropped className assertions from `MobileBottomBar.test.tsx`; switched every interaction to `useShareScope` mocks now that the local share state lifted up.

## Follow-ups (not in scope)
- Regenerate the `@goodparty_org/contracts` package so `MeetingBriefingFull` carries `briefing_id` / `meeting_time` / `meeting_timezone` and the manual augmentations in `types.ts` can go away.

Made with [Cursor](https://cursor.com)

---

> [!NOTE]
> **Medium Risk**
> Depends on coordinated gp-api deploy and env for the rewrite; missing briefing_id is guarded but share breaks if backend lags. Public UUID URLs expose PDFs to anyone with the link—expected for sharing.
> 
> **Overview**
> Replaces **client-side PDF generation** with a **Share** flow: desktop header and mobile dock open one bottom sheet (Copy link, Email, Message, Download) that all target a **marketing-domain** URL (`/api/v1/briefings/{uuid}`) proxied to **gp-api**. A new **`ShareScope`** owns a single drawer instance so header and mobile don’t mount duplicate Radix sheets.
> 
> Removes **`@react-pdf/renderer`** and the in-browser briefing PDF modules. **`Briefing`** now requires **`briefing_id`** (with a runtime guard if gp-api hasn’t shipped it yet), plus optional meeting time fields and date helpers for drawer subtext. **`next.config`** adds the briefings rewrite only when **`NEXT_PUBLIC_API_BASE`** is set.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 013ebb46335201702f0adecbd65f3ed8b6b3da5f. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Comment by oakinh** [comment]
@claude review once

File: app/shared/briefings/dateHelpers.ts [hunk 1]
```diff
@@ -61,3 +62,42 @@ export function formatShortDate(targetIso: string): string {
...
+ * be parsed (defense-in-depth for malformed artifacts).
+ */
+export function formatBriefingMeetingTime(
+  meetingTime: string | undefined,
+): string {
+  if (!meetingTime) return ''
+  const [hhRaw, mmRaw] = meetingTime.split(':')
+  const h24 = Number(hhRaw)
+  const mm = mmRaw ?? ''
+  if (!Number.isFinite(h24) || mm.length !== 2) return meetingTime
```
[the remark this answers is not in the document]
Comment (oakinh) [reply app/shared/briefings/dateHelpers.ts]: Fixed in 77a78f17 — `formatBriefingMeetingTime` now also rejects out-of-range hours (outside `0..23`) and minutes (outside `0..59`), returning the raw input verbatim as the docstring promised. The same parser exists on gp-api side and was fixed in parallel in https://github.com/thegoodparty/gp-api/pull/1657 (commit ca46821a). New `dateHelpers.test.ts` covers `25:00`, `-1:00`, `99:00`, `10:60`, `10:99`, `not-a-time`, plus the boundary cases at `00:00`, `12:00`, `23:59`.

File: gpApi/api-endpoints.ts [hunk 2]
```diff
@@ -245,9 +245,17 @@ export type APIEndpoints = {
...
+  // `goodparty.org/api/v1/briefings/{uuid}` without a second round-trip.
   'GET /v1/meetings/:date/briefing': {
     Request: { date: string }
-    Response: MeetingBriefingOutput | MeetingBriefingAwaitingDto
+    Response:
+      | (Extract<
+          MeetingBriefingOutput,
+          { briefing_status: 'briefing_ready' | 'agenda_provided_by_user' }
+        > & { briefing_id: string })
+      | MeetingBriefingAwaitingDto
```
[the remark this answers is not in the document]
Comment (oakinh) [reply gpApi/api-endpoints.ts]: Reverted in 66494fee — the augmentation in `gpApi/api-endpoints.ts` is gone and the cross-repo contract is back to its original `MeetingBriefingOutput | MeetingBriefingAwaitingDto` shape. We get the same type safety via `isFullArtifact` in `server.ts`, whose return-type annotation narrows the response to `MeetingBriefingFull & { briefing_id: string }` after a runtime `briefing_id` check — so `getBriefingBySlug` still returns the augmented `Briefing` shape that `types.ts` declares, without touching the cross-repo file. The follow-up to regenerate `@goodparty_org/contracts` so `MeetingBriefingFull` natively carries the field is still in the PR description.

File: package.json [hunk 3]
```diff
@@ -72,7 +72,6 @@
     "@radix-ui/react-toggle-group": "^1.1.11",
     "@radix-ui/react-tooltip": "^1.2.6",
     "@react-google-maps/api": "^2.20.6",
-    "@react-pdf/renderer": "^4.5.1",
     "@segment/analytics-next": "^1.81.1",
```
[the remark this answers is not in the document]
Comment (oakinh) [reply package.json:75]: Fixed in 66494fee — `npm uninstall qrcode @types/qrcode` ran; the entries are out of `package.json` and the transitive deps (`dijkstrajs`, `pngjs`, `yargs`) are out of the lockfile. The deleted `downloadBriefingPdf.tsx` really was the only consumer.

File: app/dashboard/briefings/components/detail/ShareBriefingDrawer.tsx [hunk 4]
```diff
@@ -0,0 +1,248 @@
+'use client'
+
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import { Check, Copy, Download, Link2, Mail, MessageSquare } from 'lucide-react'
```
[the remark this answers is not in the document]
Comment (oakinh) [reply app/dashboard/briefings/components/detail/ShareBriefingDrawer.tsx]: Fixed in 66494fee — added `Share2`, `Copy`, `Link2`, `Mail`, `MessageSquare` to `styleguide/components/ui/icons.tsx` using the existing `Foo as FooIcon` alias pattern, then routed all the new imports through `@styleguide`: `ShareBriefingDrawer.tsx` now consumes `CheckIcon / CopyIcon / DownloadIcon / Link2Icon / MailIcon / MessageSquareIcon`, and `DetailHeaderActions.tsx` / `MobileBottomBar.tsx` now pull `Share2Icon` from `@styleguide`. The grandfathered `MessageSquare` direct imports in those two files stay as-is per the rules explicit carve-out.

File: app/dashboard/briefings/components/detail/ShareBriefingDrawer.tsx [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+            type="button"
+            variant="default"
+            size="xSmall"
+            // Force the GoodParty primary blue so the Copy CTA is visually
+            // distinct from the muted URL pill it sits inside, regardless of
+            // any contextual button styling further up the cascade.
+            className="border-blue-600 bg-blue-600 text-white hover:border-blue-700 hover:bg-blue-700"
+            onClick={onCopyInline}
+          >
+            {copiedInline ? 'Copied' : 'Copy'}
```
[the remark this answers is not in the document]
Comment (oakinh) [reply app/dashboard/briefings/components/detail/ShareBriefingDrawer.tsx:223]: Fixed in 66494fee — the inline Copy button now uses `border-primary bg-primary text-primary-foreground hover:border-primary/90 hover:bg-primary/90`. No visual change today (the styleguide registers `--color-blue-600` as the primary token), but the button now degrades gracefully if the Tailwind default-palette override is ever restructured.

**Comment by oakinh** [comment]
delegate review

**Review by swain — APPROVED** [review]

File: app/shared/briefings/server.ts [hunk 6]
```diff
@@ -50,9 +50,17 @@ const BRIEFING_TYPE_LABEL: Record<BriefingType, string> = {
...
-  return status === 'briefing_ready' || status === 'agenda_provided_by_user'
+  if (status !== 'briefing_ready' && status !== 'agenda_provided_by_user') {
+    return false
+  }
+  // Also verify the `briefing_id` augmentation arrived from gp-api. Without
+  // this guard, a stale gp-api response (pre-share-drawer deploy) would
+  // narrow to `Briefing` and downstream `buildShareUrl` would produce a
+  // `/api/v1/briefings/undefined` link instead of failing fast.
+  const briefingId = (data as { briefing_id?: unknown }).briefing_id
+  return typeof briefingId === 'string' && briefingId.length > 0
```
[the remark this answers is not in the document]
Comment (oakinh) [reply app/shared/briefings/server.ts]: Fixed in d6cb49283 — agreed, hard 404 on the entire page is the wrong failure mode. `isFullArtifact` is back to status-only narrowing, `getBriefingBySlug` defaults `briefing_id` to `""` when the augmentation is absent, and `ShareScope` exposes a `canShare = Boolean(briefing.briefing_id)` boolean that the action surfaces consume — `DetailHeaderActions` and `MobileBottomBar` hide the Share button when `!canShare`, and the drawer itself is not even mounted in that case so a stale call site cant accidentally surface the broken URL. Negative cases locked in via `DetailHeaderActions.test.tsx` and `MobileBottomBar.test.tsx`.

File: app/dashboard/briefings/components/detail/ShareBriefingDrawer.tsx [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+  useEffect(() => {
+    return () => {
+      if (copiedIconTimerRef.current !== null) {
+        clearTimeout(copiedIconTimerRef.current)
+      }
+      if (copiedInlineTimerRef.current !== null) {
+        clearTimeout(copiedInlineTimerRef.current)
+      }
+    }
+  }, [])
```
[the remark this answers is not in the document]
Comment (oakinh) [reply app/dashboard/briefings/components/detail/ShareBriefingDrawer.tsx:87]: Fixed in d6cb49283 — added a second `useEffect` watching `open`. On `open=false` it cancels any pending timer (clearing the ref) and resets both `copiedIcon` and `copiedInline` to `false`. The original `useEffect(_, [])` unmount-cleanup stays for the full-page-teardown case. New `ShareBriefingDrawer.test.tsx` test exercises the close-and-reopen-mid-timer sequence and asserts the inline button is back to "Copy" rather than the stale "Copied" label.

**Comment by oakinh** [comment]
delegate review
